beacon_node/beacon_chain/src/migrate.rs

```diff
@@ -203,15 +211,25 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> BackgroundMigrator<E, Ho
         };
 
         // Finally, compact the database so that new free space is properly reclaimed.
-        debug!(log, "Starting database compaction");
-        if let Err(e) = db.compact() {
-            error!(
-                log,
-                "Database compaction failed";
-                "error" => format!("{:?}", e)
-            );
+        // We use a heuristic to avoid compacting too often, as it's potentially costly and puts
+        // wear on the disk.
+        let old_finalized_epoch = old_finalized_checkpoint.epoch;
+        let new_finalized_epoch = notif.finalized_checkpoint.epoch;
+        if db.compact_on_prune()
+            && (new_finalized_epoch % COMPACTION_MODULO_THRESHOLD == 0
```

Comment:
What are your thoughts on this:

```rust
old_finalized_epoch \ COMPACTION_MODULO_THRESHOLD < new_finalized_epoch \ COMPACTION_MODULO_THRESHOLD
```

This way we also detect when finality crosses a `COMPACTION_MODULO_THRESHOLD` (i.e., we detect when we skip-finalize a threshold) and we don't need the checks on the following line.

Reply:
Yes!! Perfect!